cmd/gen-go-sample/main.go

```diff
@@ -413,12 +413,25 @@ func (g *generator) handleOut(meth *descriptor.MethodDescriptorProto, valSet Sam
 	return nil
 }
 
+// prependLines adds prefix to every line in b. A line is defined as a possibly empty run
+// of non-newlines terminated by a newline character.
+// If b doesn't end with a newline, prependLines panics.
 func prependLines(b *bytes.Buffer, prefix string) {
-	rep := bytes.Replace(b.Bytes(), []byte{'\n'}, []byte("\n"+prefix), -1)
+	if b.Len() == 0 {
+		return
+	}
+	if b.Bytes()[b.Len()-1] != '\n' {
+		panic(errors.E(nil, "prependLines: must end with newline"))
```

Comment:
Is it Go-idiomatic ("Gonic"?) to panic here, or should we do the whole error dance?

Reply:
I think it's fine to panic. It's an internal function. We, not the user, are responsible for making sure there is a newline.